server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeAttribute.java

```diff
@@ -157,6 +163,10 @@ private Set<String> getValidatedRepositoryNames(DiscoveryNode node) {
         } else if (node.getAttributes().containsKey(REMOTE_STORE_CLUSTER_STATE_REPOSITORY_NAME_ATTRIBUTE_KEY)) {
             repositoryNames.add(validateAttributeNonNull(node, REMOTE_STORE_CLUSTER_STATE_REPOSITORY_NAME_ATTRIBUTE_KEY));
         }
+        if (node.getAttributes().containsKey(REMOTE_STORE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEY)) {
```

Comment:
Can routing table repository be provided without cluster state repository?

Reply:
We are keeping these settings separate to allow separate enabling/disabling if required. However, we will only allow routing table to be enabled if remote cluster state is enabled.

Reply:
But it will still be used in JoinTaskExecutor, right? For example, 2 nodes joining the cluster, one with repo details and another is without. If the feature flag is disabled, both the nodes should be able to join.

Reply:
Added feature enabled check in JoinTaskExecutor.